cyral/resource_cyral_access_rules.go

```diff
@@ -0,0 +1,292 @@
+				policyIDs := config["policy_ids"].([]interface{})
+				ids := make([]string, 0, len(policyIDs))
+				for _, policyID := range policyIDs {
+					ids = append(ids, policyID.(string))
+				}
+				accessRule.Config = &AccessRulesConfig{
+					AuthorizationPolicyInstanceIDs: ids,
+				}
+			}
+		}
+
+		accessRules = append(accessRules, accessRule)
+	}
+
+	arr.AccessRules = accessRules
+	return nil
+}
+
+var ReadRepositoryAccessRulesConfig = ResourceOperationConfig{
+	Name:       "RepositoryAccessRulesRead",
+	HttpMethod: http.MethodGet,
+	CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+		return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+			c.ControlPlane,
+			d.Get("repository_id").(string),
+			d.Get("user_account_id").(string),
+		)
+	},
+	NewResponseData: func(_ *schema.ResourceData) ResponseData {
+		return &AccessRulesResponse{}
+	},
+}
+
+func resourceRepositoryAccessRules() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manage access rules",
+		CreateContext: CreateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesCreate",
+				HttpMethod: http.MethodPost,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					d.SetId(
+						marshalComposedID(
+							[]string{
+								d.Get("repository_id").(string),
+								d.Get("user_account_id").(string),
+							},
+							"/",
+						),
+					)
+					return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Get("user_account_id").(string),
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &AccessRulesResource{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &AccessRulesResponse{}
+				},
+			},
+			ReadRepositoryAccessRulesConfig,
+		),
+		ReadContext: ReadResource(ReadRepositoryAccessRulesConfig),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Get("user_account_id").(string),
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &AccessRulesResource{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &AccessRulesResponse{}
+				},
+			},
+			ReadRepositoryAccessRulesConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+
+					idPieces, err := unmarshalComposedID(d.Id(), "/", 2)
+					if err != nil {
+						panic("Failed to unmarshal access rules ID")
```

Comment:
```suggestion
						panic("Failed to unmarshal access rules ID: " + err.Error())
```
I know we don't expect this to panic, but maybe it would be nice to include the error information, just in case.

Reply:
Good idea. Done.